packages/core/integration-test/integration-test.js

```diff
@@ -520,47 +524,94 @@ const doTest = runner => {
         command: 'execute',
         method: 'resources.requestfunc.list.operation.perform',
         logExtra: {
-          app_cli_id: 666
-        }
+          app_cli_id: 666,
+        },
       };
-      return runner(event).then(response => {
+      return runner(event).then((response) => {
         should.exist(response.results);
       });
     });
 
     it('should not leave leftover env vars', () => {
       const event = {
         environment: {
-          _ZAPIER_ONE_TIME_SECRET: 'foo'
+          _ZAPIER_ONE_TIME_SECRET: 'foo',
         },
-        method: 'resources.env.list.operation.perform'
+        method: 'resources.env.list.operation.perform',
       };
       return runner(event)
-        .then(response => {
+        .then((response) => {
           response.results.length.should.eql(1);
           response.results[0].key.should.eql('_ZAPIER_ONE_TIME_SECRET');
           response.results[0].value.should.eql('foo');
 
           delete event.environment;
           return runner(event);
         })
-        .then(response => {
+        .then((response) => {
           response.results.length.should.eql(0);
         });
     });
 
+    describe('hang detection', () => {
+      let originalEnv;
+
+      before(() => {
+        originalEnv = process.env;
+        process.env = {
+          ...originalEnv,
+          LOGGING_ENDPOINT: `${mocky.FAKE_LOG_URL}/input`,
+          LOGGING_TOKEN: 'fake-token',
+        };
+      });
+
+      after(() => {
+        process.env = originalEnv;
+      });
+
+      beforeEach(() => {
+        mocky.clearLogs();
+      });
+
+      afterEach(() => {
+        mocky.clearLogs();
+      });
+
+      it('should end logger even a callback runs after lambda handler returns', async () => {
+        mocky.mockLogServer();
+
+        const event = {
+          method: 'resources.bad_callback.create.operation.perform',
+        };
+        const response = await runner(event);
+        response.results.message.should.eql('ok');
+        should.not.exist(process.teapot);
+
+        // Hopefully long enough for the callback to complete
+        await sleep(2000);
+
+        process.teapot.should.eql("I'm a teapot!");
+
+        const logs = mocky.getLogs();
+        logs.length.should.eql(1);
+
+        const log = logs[0];
+        log.message.should.startWith(`418 GET https://httpbin.`);
```

Comment:
Changed this from `should.eql` to `should.startWith` because the URL in the log message was censored. Apparently, GitHub Actions has a `zapier` in one of the environment variables and our log sanitizer uses to censor `https://httpbin.zapier-tooling.com` into `https://httpbin.:censored:6:e19648f405:-tooling.com`.